src/changelog.ts

```diff
@@ -192,42 +187,137 @@ function stringifyLinkReferenceDefinitions(
   //
   // If there have not been any releases yet, the repo URL is used directly as
   // the link definition.
-  const unreleasedLinkReferenceDefinition = `[${unreleased}]: ${
-    hasReleases
-      ? getCompareUrl(repoUrl, `${tagPrefix}${latestSemverVersion}`, 'HEAD')
-      : withTrailingSlash(repoUrl)
-  }`;
+  const unreleasedLinkReferenceDefinition =
+    getUnreleasedLinkReferenceDefinition(
+      repoUrl,
+      tagPrefix,
+      releases,
+      versionBeforePkgRename,
+      tagPrefixBeforePkgRename,
+    );
 
   // The "previous" release that should be used for comparison is not always
   // the most recent release chronologically. The _highest_ version that is
   // lower than the current release is used as the previous release, so that
   // patch releases on older releases can be accomodated.
-  const releaseLinkReferenceDefinitions = releases
-    .map(({ version }) => {
-      let diffUrl;
-      if (version === chronologicalVersions[chronologicalVersions.length - 1]) {
-        diffUrl = getTagUrl(repoUrl, `${tagPrefix}${version}`);
+  // by default tag prefix from new package will be used
+  const releaseLinkReferenceDefinitions = getReleaseLinkReferenceDefinitions(
+    repoUrl,
+    tagPrefix,
+    releases,
+    versionBeforePkgRename,
+    tagPrefixBeforePkgRename,
+  ).join('\n');
+  return `${unreleasedLinkReferenceDefinition}\n${releaseLinkReferenceDefinitions}${
+    releases.length > 0 ? '\n' : ''
+  }`;
+}
+
+/**
+ * Get a string of unreleased link reference definition.
+ *
+ * @param repoUrl - The URL for the GitHub repository.
+ * @param tagPrefix - The prefix used in tags before the version number.
+ * @param releases - The releases to generate link definitions for.
+ * @param [versionBeforePkgRename] - A version string of the package before being renamed.
+ * @param [tagPrefixBeforePkgRename] - A tag prefix string of the package before being renamed.
+ * @returns A unreleased link reference definition string.
+ */
+function getUnreleasedLinkReferenceDefinition(
+  repoUrl: string,
+  tagPrefix: string,
+  releases: ReleaseMetadata[],
+  versionBeforePkgRename?: string,
+  tagPrefixBeforePkgRename?: string,
```

Comment:
This would have the advantage of not breaking if `exactOptionalPropertyTypes` is turned on.

Reply:
I get what you mean and that's how it is been called now. But when we look at the function utility, it can work even without these parameters. We need these parameters only in case of package is renamed. That way, I think, it's better them to be optional.